web/src/admin/roles/RelatedRoleList.ts

```diff
@@ -0,0 +1,185 @@
+import "#components/ak-status-label";
+import "#elements/buttons/SpinnerButton/index";
+import "#elements/forms/DeleteBulkForm";
+import "#elements/forms/HorizontalFormElement";
+import "#elements/forms/ModalForm";
+import "@patternfly/elements/pf-tooltip/pf-tooltip.js";
+
+import { DEFAULT_CONFIG } from "#common/api/config";
+
+import { Form } from "#elements/forms/Form";
+import { PaginatedResponse, Table, TableColumn } from "#elements/table/Table";
+import { SlottedTemplateResult } from "#elements/types";
+
+import { RbacApi, Role, User } from "@goauthentik/api";
+
+import { msg, str } from "@lit/localize";
+import { html, nothing, TemplateResult } from "lit";
+import { customElement, property, state } from "lit/decorators.js";
+import { ifDefined } from "lit/directives/if-defined.js";
+
+@customElement("ak-role-related-add")
+export class RelatedRoleAdd extends Form<{ roles: string[] }> {
+    @property({ attribute: false })
+    user?: User;
+
+    @state()
+    rolesToAdd: Role[] = [];
+
+    getSuccessMessage(): string {
+        return msg("Successfully added user to role(s).");
+    }
+
+    async send(data: { roles: string[] }): Promise<unknown> {
+        await Promise.all(
+            data.roles.map((role) => {
+                if (!this.user) return;
+                return new RbacApi(DEFAULT_CONFIG).rbacRolesAddUserCreate({
+                    uuid: role,
+                    userAccountSerializerForRoleRequest: {
+                        pk: this.user.pk,
+                    },
+                });
+            }),
+        );
+        return data;
+    }
+
+    renderForm(): TemplateResult {
+        return html`<ak-form-element-horizontal label=${msg("Roles to add")} name="roles">
+            <div class="pf-c-input-group">
+                <ak-user-role-select-table
+                    .confirm=${(items: Role[]) => {
+                        this.rolesToAdd = items;
+                        this.requestUpdate();
+                        return Promise.resolve();
+                    }}
+                >
+                    <button slot="trigger" class="pf-c-button pf-m-control" type="button">
+                        <pf-tooltip position="top" content=${msg("Add role")}>
+                            <i class="fas fa-plus" aria-hidden="true"></i>
+                        </pf-tooltip>
+                    </button>
+                </ak-user-role-select-table>
+                <div class="pf-c-form-control">
+                    <ak-chip-group>
+                        ${this.rolesToAdd.map((role) => {
+                            return html`<ak-chip
+                                removable
+                                value=${ifDefined(role.pk)}
+                                @remove=${() => {
+                                    const idx = this.rolesToAdd.indexOf(role);
+                                    this.rolesToAdd.splice(idx, 1);
+                                    this.requestUpdate();
```

Comment:
Immutable approach, something along the lines of...
```suggestion
                                    this.rolesToAdd = this.rolesToAdd.filter(($role) => $role === role);
```